serverlessworkflow/modules/ROOT/pages/eventing/working-with-callbacks.adoc

```diff
@@ -1,26 +1,43 @@
 
+From a Serverless Workflow perspective, asynchronous service indicates that the control is returned to the caller immediately without waiting for the action to be completed. Once the action is completed, a link:{cloud_events_url}[CloudEvent] is published to resume the Serverless Workflow.
 
-This guide discuss the link:{spec_doc_url}#Callback-State[Callback State]. This is a state that performs an action and waits for an event produced as a result of that action to resume the workflow. So it is suitable to perform __fire&wait-for-result__ operations. The action will normally be an external service invocation. This service is expected to be async.
+For the Serverless Workflow to identify the published CloudEvent it is waiting for, the external service developer includes the process instance ID in the CloudEvent header or use the xref:event-correlation-with-workflows[Serverless Workflow correlation mechanism]. 
 
-From the flow perspective, async means that the control will be returned to the caller almost immediately, without waiting for the actual operation to be finished. Once it ends, a link:{cloud_events_url}[Cloud Event] will be published to resume the flow.
+The Serverless Workflow correlation mechanism document focuses on the former mechanism that is based on the fact that each Serverless Workflow instance contains a unique identifier, which is generated automatically.
 
-For the flow to know this published event is the one it is waiting for, the external service developer might include the Kogito process instance id in the event header or, preferably, use the xref:event-correlation-with-workflows[Serverless Workflow correlation mechanism]. This guide will focus on the former mechanism, which is based on the fact that every Kogito Workflow instance has an automatically generated and unique identifier.
+[[ref-example-callback]]
+== Example of Callback state
 
-== Error callback example 
+To understand the Callback state, see the link:{kogito_sw_examples_url}/serverless-workflow-callback-quarkus[`serverless-workflow-callback-quarkus`] example application in GitHub repository. 
 
-The workflow in the link:{kogito_sw_examples_url}/serverless-workflow-callback-quarkus[Serverless Workflow Callback Quarkus] example illustrates the callback feature. Once started (this workflow does not have an initial model), it publishes a Cloud Event of type `resume` and waits for a Cloud Event of `wait` type.
+The Serverless Workflow in the `serverless-workflow-callback-quarkus` example application does not consist of an initial model. However, once the workflow is started, it publishes a CloudEvent of the `resume` type and waits for a CloudEvent, which contains the type `wait`.
```

Comment:
This is also changing the meaning. Im trying to say that the initial model of the workflow is empty. Now it is  saying that the flow is not an initial model

Reply:
Original sentence: Once started (this workflow does not have an initial model), it publishes a Cloud Event of type `resume` and waits for a Cloud Event of `wait` type.

Modified sentence:  The Serverless Workflow in the `serverless-workflow-callback-quarkus` example application does not consist of an initial model. However, once the workflow is started, it publishes a CloudEvent of the `resume` type and waits for a CloudEvent, which contains the type `wait`.

The modified sentence is not stating that the "flow is not an initial model", but stating the same as stated in the original sentence "Serverless Workflow in the `serverless-workflow-callback-quarkus` example application does not consist of an initial model". 

However, if you meant that the initial model is empty, then we can change this sentence as follows:

The Serverless Workflow in the `serverless-workflow-callback-quarkus` example application **consists of an empty initial model.** However, once the workflow is started, it publishes a CloudEvent of the `resume` type and waits for a CloudEvent, which contains the type `wait`.

Reply:
https://www.dictionary.com/browse/consist. does not has a clear different meaning that does not have
For me the sentece should be
The initial model of  the serverless-workflow-callback-quarkus example application is empty, Once the workflow is started, it publishes a CloudEvent of the resume type and waits for a CloudEvent of the type wait.

Reply:
Done, changed the sentence as follows:

The initial model of the `serverless-workflow-callback-quarkus` example application is empty. Therefore, once the workflow is started, it publishes a CloudEvent of the `resume` type and waits for a CloudEvent, which contains the type `wait`.

Reply:
I do not understand why do we need the the `therefore`. This are two unrelated sentences. We are stating that the flow initial model is empty and in the next sentence, which is independent of this fact, we state what the workflow does.

Reply:
Removed, therefore.